_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,179 @@
+Now we can run our build. As result of we should see our **artifacts** in *"Artifact"* tab. 
+ 
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_result.png" alt="Build Result" title="Build Result" class="img img-responsive style-screengrab">
+
+## Release Image
+
+Our code is no longer needed. We have all we need to build the docker image. Now we have create our image and **release** with the right version. To simplify our example we are going to use current build number to define an image version.
+Next we will generate file with this version. This file give us possibility to pass information about version to the next steps.  
+
+First, let's take a closer look at `Dockerfile`. We copy to the image all libraries and scripts from `simple_application.tar`. This ***tar*** file already contains all required content, so all we have to do is to use `ADD`. It will automatically *untar* all files in the image. Next we expose port of our HTTP API and we define how we launch our application by adding `ENTRYPOINT` command.  
+
+For example our `Dockerfile` can look like as follows: 
+
+```
+FROM java:8
+
+ADD simple_application.tar .
+
+EXPOSE 4567
+
+ENTRYPOINT ["/simple_application/bin/simple_application"]
+```
+
+Definition of the build configuration is very simple. In the general settings we just define new artifact: `image.version`. We will generate this file in a build step and it will contain an docker image version.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_config.png" alt="Release Configuration" title="Release Configuration" class="img img-responsive style-screengrab">
+
+Without **artifact** we won't be able to build any image. We have to tell our build how to find **artifacts** from **Build** phase. We can do that by defining **Artifact Dependency** in **TeamCity**. We just have to choose build configuration and define files which we want to use in this build.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_artifacts.png" alt="Release Artifact Dependency" title="Release Artifact Dependency" class="img img-responsive style-screengrab">
+
+And finally we have to trigger this build automatically when we pass all **tests** in the previous step. By introducing **Finish Build Trigger** we can start this build just after **TeamCity** successfully finish building the application.  
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_trigger.png" alt="Release Trigger" title="Release Trigger" class="img img-responsive style-screengrab">
+
+Now we are ready for release. Three build steps must be introduced: **Build Image**, **Push Image to Repository** and **Save version**. 
+This time we will use different runner type: **Command Line**. We can execute shell script on a build agent. Because we already installed **Docker** on our build agent we can use `docker` command in our shell script.
```

Comment:
This time we will use a different runner type

 We can execute a shell script on a build agent. Because we already installed **Docker** on our build agent, we can use the `docker` command in our shell script.